src/confluent_kafka/admin/__init__.py

```diff
@@ -1258,3 +1274,39 @@ def delete_records(self, topic_partition_offsets, **kwargs):
 
         super(AdminClient, self).delete_records(topic_partition_offsets, f, **kwargs)
         return futmap
+
+    def elect_leaders(self, election_type, partitions, **kwargs):
+        """
+        Perform Preferred or Unclean elections for,
+        all the specified topic partitions.
+
+        :param election_type: ElectionType - The type of election to perform.
+        :param partitions: List[TopicPartition] - The topic partitions to perform
+               the election on.
+        :param float request_timeout: The overall request timeout in seconds,
+                     including broker lookup, request transmission, operation time
+                     on broker, and response. Default: `socket.timeout.ms*1000.0`
+        :param float operation_timeout: The operation timeout in seconds,
+                     controlling how long the 'elect_leaders' request will block
+                     on the broker waiting for the election to propagate
+                     in the cluster. A value of 0 returns immediately.
+                     Default: `socket.timeout.ms/1000.0`
+
+        :returns: A dict of futures keyed by the :class:`.TopicPartition`.
+                  The future result() method returns None or
+                  raises KafkaException.
+
+        :rtype: rtype: dict[TopicPartition, future]
+
+        :raises KafkaException: Operation failed locally or on broker.
+        :raises TypeError: Invalid input type.
+        :raises ValueError: Invalid input value.
+        """
+
+        AdminClient._check_elect_leaders(election_type, partitions)
+
+        f, futmap = AdminClient._make_futures_v2(partitions, _TopicPartition, AdminClient._make_futmap_result)
```

Comment:
Input topic partitions can be None as well. In this case, we have to make future in different way. Please check https://github.com/confluentinc/confluent-kafka-python/blob/master/src/confluent_kafka/admin/__init__.py#L1148-L1152 for reference on how to make that work.